arangod/Agency/Agent.cpp

```diff
@@ -1960,10 +1961,16 @@ void Agent::emptyCbTrashBin() {
     _callbackLastPurged = std::chrono::steady_clock::now();
   }
 
-  LOG_TOPIC(DEBUG, Logger::AGENCY) << "unobserving: " << envelope->toJson();
-
-  // Best effort. Will be retried anyway
-  auto wres = write(envelope);
+  // Best effort. Will be retried otherwise.
+  LOG_TOPIC(DEBUG, Logger::AGENCY) << "scheduling unobserve: " << envelope->toJson();
+  auto* scheduler = SchedulerFeature::SCHEDULER;
+  std::shared_ptr<Agent> agent = this->shared_from_this();
+  if (scheduler != nullptr) {
+    scheduler->queue(
+      RequestPriority::LOW, [agent, envelope](bool) {
```

Comment:
for devel and 3.5, this could be
```suggestion
      RequestPriority::LOW, [agent = std::move(agent), envelope = std::move(envelope)](bool) {
```